### PR TITLE
JAMES-2605 remove hardcore parallelism from rabbitmq MailQueue

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

File: server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java [hunk 1]
```diff
@@ -101,23 +98,22 @@ public boolean hasNext() {
...
-            .thenApply(this::allSlicesStartingAt))
-            .map(slice -> browseSlice(queueName, slice), FluentFutureStream::unboxFluentFuture);
+    Stream<EnqueuedItemWithSlicingContext> browseReferences(MailQueueName queueName) {
+        return OptionalUtils.toStream(browseStartDao.findBrowseStart(queueName).join())
+            .flatMap(this::allSlicesStartingAt)
+            .flatMap(slice -> browseSlice(queueName, slice));
     }
 
-    private CompletableFuture<Mail> toMailFuture(EnqueuedItemWithSlicingContext enqueuedItemWithSlicingContext) {
+    private Mail toMailFuture(EnqueuedItemWithSlicingContext enqueuedItemWithSlicingContext) {
```
Comment (rouazana) [inline server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueBrowser.java]: should rename this method

**Comment by rouazana** [comment]
```
[585643d1aa161763f209678e0fa17aec280bb66e] [ERROR] COMPILATION ERROR : 
[585643d1aa161763f209678e0fa17aec280bb66e] [INFO] -------------------------------------------------------------
[585643d1aa161763f209678e0fa17aec280bb66e] [ERROR] /james-project/backends-common/cassandra/src/test/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionDAOTest.java:[41,18] constructor CassandraSchemaVersionDAO in class org.apache.james.backends.cassandra.versions.CassandraSchemaVersionDAO cannot be applied to given types;
[585643d1aa161763f209678e0fa17aec280bb66e]   required: com.datastax.driver.core.Session,org.apache.james.backends.cassandra.utils.CassandraUtils
[585643d1aa161763f209678e0fa17aec280bb66e]   found: com.datastax.driver.core.Session
[585643d1aa161763f209678e0fa17aec280bb66e]   reason: actual and formal argument lists differ in length
```

**Comment by rouazana** [comment]
> What are the performance benefits?

The performance benefits are negative ( -30%). But the expectation are better, as we have no more 20% of errors!

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by hoangdat** [comment]
Build started sha1 is merged.

**Comment by hoangdat** [comment]
Build finished. No test results found.

**Comment by hoangdat** [comment]
Build triggered. sha1 is merged.

**Comment by rouazana** [comment]
it conflicts, can you please rebase it?

**Comment by mbaechler** [comment]
It has been replaced with a Reactor implementation https://github.com/linagora/james-project/pull/1942

**Pull request closed without merging**